packages/aws-cdk/lib/init-templates/app/dotnet/src/HelloCdk/HelloConstruct.cs

```diff
@@ -0,0 +1,29 @@
+using Amazon.CDK;
+using Amazon.CDK.AWS.IAM;
+using Amazon.CDK.AWS.S3;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace HelloCdk
+{
+    public class HelloConstruct : Construct
+    {
+        readonly IList<Bucket> _buckets = new List<Bucket>();
+
+        public HelloConstruct(Construct parent, string id, HelloConstructProps props) : base(parent, id)
+        {
+            foreach (int i in Enumerable.Range(0, props.BucketCount))
```

Comment:
move to GenerateBuckets method that returns an IEnumerable<Bucket>. Chnage line to _buckets = GenerateBuckets(props.BucketCount);

private IEnumerable<Bucket> GenerateBuckets() => Enumerable.Range(0, number).Select(number => new Bucket(this, $"Bucket{i}", new BucketProps())));

Reply:
✔️ (simplified in a slightly different way)